upload/system/library/Packetery/Vendor/AddEditPage.php

```diff
@@ -0,0 +1,340 @@
+
+		foreach ($weightRules as $rule) {
+			$maxWeight = $rule['max_weight'];
+
+			if (isset($uniqueWeights[$maxWeight])) {
+				return true; // Duplicate weight found
+			}
+
+			$uniqueWeights[$maxWeight] = true;
+		}
+
+		return false; // No duplicate weights found
+	}
+
+	/**
+	 * Session will contain invalid form data for form pre-filling
+	 *
+	 * @param array $formData
+	 *
+	 * @return void
+	 */
+	private function saveAddVendorFormDataToSession(array $formData) {
+		$this->session->data['vendor_add_form_data'] = $formData;
+	}
+
+	/**
+	 * @return array
+	 */
+	private function getAddVendorFormDataFromSession() {
+		if (!isset($this->session->data['vendor_add_form_data'])) {
+			return [];
+		}
+
+		$formData = $this->session->data['vendor_add_form_data'];
+		unset($this->session->data['vendor_add_form_data']);
+
+		return $formData;
+	}
+
+	/**
+	 * @param array $formData
+	 *
+	 * @return array
+	 */
+	public function getFormDefaults(array $formData = []) {
+
+		if (empty($formData)) {
+			$formData = $this->getAddVendorFormDataFromSession();
+		}
+		$prefilledData = [];
+		foreach ($formData as $key => $value) {
+			if (in_array($key, ['vendor', 'country', 'vendor_group', 'cart_name', 'free_shipping_limit', 'is_enabled', 'weight_rules'])) {
+				$prefilledData[$key] = $value;
+			}
+		}
+
+		return $prefilledData;
+	}
+
+	/**
+	 * @param array $weightRules
+	 *
+	 * @return array
+	 */
+	private function sortWeightRulesByMaxWeight(array $weightRules) {
+		usort($weightRules, static function ($a, $b) {
+			return (float)$a['max_weight'] - (float)$b['max_weight'];
+		});
+
+		return $weightRules;
+	}
+
+	/**
+	 * @param array $formData
+	 *
+	 * @return void
+	 */
+	public function saveVendorWithWeightRules(array $formData) {
+		$isCarrier = is_numeric($formData['vendor']);
+
+		if ($formData['vendor'] === 'zpoint'){
+			$formData['vendor'] = '';
+		}
+
+		$vendor = [
+			'id'                  => null,
+			'carrier_id'          => $isCarrier ? (int)$formData['vendor'] : null,
+			'carrier_name_cart'   => $formData['cart_name'],
+			'country'             => $isCarrier ? null : $formData['country'],
+			'group'               => $isCarrier ? null : $formData['vendor'],
+			'free_shipping_limit' => $formData['free_shipping_limit'],
+			'is_enabled'          => (int)$formData['is_enabled'],
+		];
+
+		$newVendorId = $this->vendorRepository->saveVendor($vendor);
+		if (!$newVendorId) {
+			return;
+		}
+
+		$vendorPrices = [];
+		if ($formData['weight_rules'] && is_array($formData['weight_rules'])) {
+			foreach ($formData['weight_rules'] as $weightRule) {
+				$vendorPrices[] = [
+					'id'         => null,
+					'vendor_id'  => $newVendorId,
+					'max_weight' => (float)$weightRule['max_weight'],
+					'price'      => (float)$weightRule['price'],
+				];
+			}
+			if ($vendorPrices) {
+				$this->vendorRepository->insertVendorPrices($vendorPrices);
+			}
+		}
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getTranslations() {
```

Comment:
Proč potřebuješ sestavit pole s překlady a předávat je šabloně ?  Twig si natahuje překlady automaticky.  V šabloně by ti mělo stačit zavolat např.  {{vendor_add_vendor_label}} a dostaneš překlad ze slovníku.